File: 418.py
from typing import List

class Solution:
    def wordsTyping(self, sentence: List[str], rows: int, cols: int) -> int:
        senLen = 0
        for sen in sentence: senLen += len(sen) + 1

        index = 0
        res = 0
        for i in range(rows):
            cur = cols
            while cur > 0 and cur >= len(sentence[index]):
                cur -= len(sentence[index])
                index += 1
                if cur > 0: cur -=1
                if index == len(sentence):
                    res += cur / senLen + 1
                    cur %= senLen
                    index = 0

        return res

File: java/src/418.java
class Solution {
    public int wordsTyping(String[] sentence, int rows, int cols) {
        int len = sentence.length;
        int result = 0;
        int curr = 0;
        int idx = 0;
        int l = 0;
        for (String s : sentence) {
            l = l + s.length()+1;
        }
        for (int i = 0; i < rows; i++) {
            curr = cols;
            while (curr > 0 && curr >= sentence[idx].length()) {
                curr = curr - sentence[idx].length();
                idx++;
                if (curr > 0) {
                    curr--;
                }
                if (idx == len) {
                    result += curr / l+1;
                    curr = curr % l;
                    idx = 0;

                }

            }
        }
        return result;
    }
}